### ISSUE TITLE
Inclusive vs Exclusive Duration for File Stats

### ISSUE THREAD

**Issue by tusharb86** [MicrosoftDocs/cpp-docs#1770]
Hi there - this tool has been super helpful to understand what header files to include in a precompiled header for our team! Thanks :)

I'm trying to understand what the "Inclusive Duration" and "Exclusive Duration" fields represent in the Files view. The numbers don't seem to align with reality on the build time of the single native project I'm profiling. For example, the Inclusive Duration shows ~7k seconds and the Exclusive Duration shows ~600 seconds, but the build was about 100 seconds only. Any details to understand these numbers would be very helpful!

Cheers,
Tushar

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d3d636dc-5117-ad56-a174-8c26874632ea
* Version Independent ID: ae39b34d-6746-afcd-d940-c26b6940f5b4
* Content: [Get started with C++ Build Insights](https://docs.microsoft.com/en-us/cpp/build-insights/get-started-with-cpp-build-insights?view=vs-2019#feedback)
* Content Source: [docs/build-insights/get-started-with-cpp-build-insights.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/build-insights/get-started-with-cpp-build-insights.md)
* Product: **visual-cpp**
* Technology: **cpp-tools**
* GitHub Login: @kevcadieux
* Microsoft Alias: **kevca**

**Comment by kevcadieux** [comment]
Hi Tushar,

In the Files view, Inclusive Duration represents the amount of time spent parsing a given header and all child headers it includes. In contrast, the Exclusive Duration represents only the amount of time spent parsing the header itself, and not the child headers it includes.

Header parsing typically occurs on multiple threads, so adding up all Inclusive Durations will usually produce a result that is larger than the wall-clock time of the build.

Does this answer your questions?

**Comment by tusharb86** [comment]
Yes - this is super helpful! It would be great to add these details in a FAQ on this page:
https://docs.microsoft.com/en-us/cpp/build-insights/get-started-with-cpp-build-insights?view=vs-2019

**Comment by TylerMSFT** [comment]
We are in the process of moving away from GitHub issues for C++ documentation (we still accept pull requests). For details, please see [Announcing a new way to give feedback on Microsoft Learn](https://techcommunity.microsoft.com/t5/microsoft-learn-blog/announcing-a-new-way-to-give-feedback-on-microsoft-learn/ba-p/4027635) This issue will be closed after it is copied to an internal tracking system.
